source/MaterialXGenOgsXml/OgsXmlGenerator.cpp

```diff
@@ -138,75 +142,57 @@ namespace
         }
     }
 
-    void xmlSetProperty(pugi::xml_node& prop, const string& name, const string& variable, const string& flags = EMPTY_STRING)
+    void xmlSetProperty(pugi::xml_node& prop, const string& name, const string& variable, const string& defaultFlags = EMPTY_STRING)
     {
         prop.append_attribute(NAME) = variable.c_str();
-        auto semantic = OGS_SEMANTICS_MAP.find(name);
+        const auto semantic = OGS_SEMANTICS_MAP.find(name);
         if (semantic != OGS_SEMANTICS_MAP.end())
         {
             prop.append_attribute(SEMANTIC) = semantic->second;
         }
-        auto flag = OGS_FLAGS_MAP.find(name);
+        const auto flag = OGS_FLAGS_MAP.find(name);
         if (flag != OGS_FLAGS_MAP.end())
         {
             prop.append_attribute(FLAGS) = flag->second.c_str();
         }
-        else if (!flags.empty())
+        else if (!defaultFlags.empty())
         {
-            prop.append_attribute(FLAGS) = flags.c_str();
+            prop.append_attribute(FLAGS) = defaultFlags.c_str();
         }
     }
 
-    void xmlAddProperties(pugi::xml_node& parent, const VariableBlock& block, const string& flags = EMPTY_STRING)
+    void xmlAddProperties(pugi::xml_node& parent, const VariableBlock& block, const string& defaultFlags = EMPTY_STRING)
     {
         for (size_t i = 0; i < block.size(); ++i)
         {
-            const ShaderPort* p = block[i];
-            if (p->getName() == "Pw")
```

Comment:
Pw is no longer forced to come first because that makes the order of vertex data values mismatch between the XML definition and the root function arguments.